lm_eval/api/task.py

```diff
@@ -348,20 +356,71 @@ def doc_to_text(self, doc):
     def doc_to_target(self, doc):
         pass
 
-    def build_all_requests(self, limit=None, rank=None, world_size=None) -> None:
+    def build_all_requests(
+        self,
+        task_name,
+        limit=None,
+        rank=None,
+        world_size=None,
+        cache_requests=False,
+        rewrite_requests_cache=False,
+    ) -> None:
         """Build a set of Instances for a task, and store them in task.instances"""
+
+        # used with caching
+        og_limit = limit
+
+        cache_key = f"requests-{task_name}"
```

Comment:
as mentioned in the other comment--perhaps we should base64-encode this or something, so that the odds of file collisions / deleting undesired files is minimized.

Reply:
I've added this in the recent commit, lmk if it looks good to you! Can see how it potentially wasn't safe enough